main.js

```diff
@@ -92,41 +92,67 @@ function removeError(){
 
 // Creating html registration form
 function createLoginForm() {
-    const div_form_container = document.createElement("div");
-    div_form_container.setAttribute("class", "form_container")
-
-    const form = div_form_container.appendChild(document.createElement("form"));
-    form.setAttribute("class", "form");
-    form.setAttribute("id", "login")
-
-    const title = form.appendChild(document.createElement("h1"));
-    title.textContent = "Login form";
-
-    const div_form_userName = form.appendChild(document.createElement("div"));
-    div_form_userName.setAttribute("class", "form_input");
-    div_form_userName.setAttribute("id", "form_username");
-    div_form_userName.innerHTML = '<label for="username">Username:</label>' +
-    '<input type="text" name="username" id="username" placeholder="username">';
-
-    const div_form_password = form.appendChild(document.createElement("div"));
-    div_form_password.setAttribute("class", "form_input");
-    div_form_password.setAttribute("id", "form_password");
-    div_form_password.innerHTML = '<label for="password">Password:</label>' +
-    '<input type="password" name="password" id="password" placeholder="password">';
-
-    const div_button = form.appendChild(document.createElement("div"));
-    div_button.setAttribute("class", "button");
-    div_button.innerHTML = '<button id="login_btn">login</button>';
-    
-    document.body.append(div_form_container);
-    return div_form_container;
-}
-
+    document.body.append(createFormContainer());
 
+    const parentElement = document.getElementById("login");
+    parentElement.appendChild(createTitle("Login form"));
+    parentElement.appendChild(createInputUsername());
+    parentElement.appendChild(createInputPassword());
+    parentElement.appendChild(createButton("login_btn")); 
+}
 
+function createFormContainer(){
+    const div_form_container = createBaseElement({className: "form_container", id: ""});
+    div_form_container.appendChild(createBaseElement({element: "form", className: "form", id: "login"}));
+    return div_form_container;
+}
 
+function createInputUsername(){
+    const username = createBaseElement({className: "form_input", id: "form_username"});
+    username.appendChild(createLabel("username", "Username"));
+    username.appendChild(createInput("username"));
+    return username;
+}
 
+function createInputPassword(){
```

Comment:
no code duplication

It would be better to combine these functions (`createInputUsername`, `createInputPassword`) into `createInputField`

Reply:
Done